chord_metadata_service/chord/workflows/wdls/experiments_json_with_files.wdl

```diff
@@ -0,0 +1,385 @@
+        File results_post_drs = write_drs_responses_to_file.results_post_drs
+        File processed_drs_responses = parse_txt.processed_drs_responses
+        File final_updated_json = update_experiment_json.final_updated_json
+    }
+}
+
+task prepare_download_list {
+    input {
+        File json_document
+        String directory
+    }
+    command <<<
+    python3 -c "
+import json
+import os
+
+directory = '~{directory}'
+
+with open('~{json_document}', 'r') as file:
+    data = json.load(file)
+
+# List of allowed protocols
+allowed_protocols = ['http://', 'https://', 'ftp://', 'ftps://']
+
+def is_allowed_protocol(url):
+    return any(url.startswith(protocol) for protocol in allowed_protocols)
+
+download_list = []
+path_list = []
+for experiment in data.get('experiments', []):
+    for result in experiment.get('experiment_results', []):
+        url = result.get('url', '')
+        filename = result.get('filename', '')
+        if url and is_allowed_protocol(url):
+            download_info = {'url': url, 'filename': filename}
+            download_list.append(download_info)
+            # Process indices
+            for index in result.get('indices', []):
+                index_url = index.get('url', '')
+                index_format = index.get('format', '')
+                if index_url and is_allowed_protocol(index_url):
+                    index_info = {'url': index_url, 'filename': filename + '.' + index_format.lower()}
+                    download_list.append(index_info)
+        else:
+            file_found = False
+            for root, dirs, files in os.walk(directory):
+                if filename in files:
+                    file_found = True
+                    file_path = os.path.join(root, filename)
+                    path_list.append({'filename': filename, 'url': file_path})
+                    break
+            if not file_found:
+                print(f'File not found for {filename}')
+
+with open('download_list.json', 'w') as outfile:
+    json.dump(download_list, outfile, indent=4)
+
+with open('path_list.json', 'w') as file:
+    json.dump(path_list, file, indent=4)
+    "
+    >>>
+    output {
+        File urls_json = "download_list.json"
+        File path_list = "path_list.json"
+    }
+}
+
+task prepare_for_download {
+    input {
+        File json_document
+    }
+
+    command <<<
+    python3 -c "
+import json
+
+with open('~{json_document}', 'r') as file:
+    data = json.load(file)
+
+download_list = [{'url': item['url'], 'filename': item['filename']} for item in data]
+
+print(json.dumps(download_list))
+    "
+    >>>
+    output {
+         Array[Map[String, String]] download_list = read_json(stdout())
+    }
+}
+
+task download_file {
+    input {
+        String url
+        String filename
+        String directory
+        String access_token
+        Boolean validate_ssl
+    }
+
+    command <<<
+    curl -L "~{url}" -o "~{filename}" -H "Authorization: Bearer ~{access_token}"
```

Comment:
I think if files already have URLs in experiment results, we shouldn't ingest them into DRS, since they're public. Also, be very careful when sending the access token: this code sends the access token to whatever server is listed in the experiment result file, which leaks the token to the corresponding third-party servers! Bearer tokens should never leave "Bento-land" unless we have explicitly trusted partners in the future.